op-batcher/batcher/txmgr.go

```diff
@@ -19,25 +18,48 @@ import (
 
 const networkTimeout = 2 * time.Second // How long a single network request can take. TODO: put in a config somewhere
 
+// ExternalTxManager is a minimal interface for the [TransactionManager]
+// that is a subset of [txmgr.SimpleTxManager] functionality
+//
+//go:generate mockery --name ExternalTxManager --output ./mocks
```

Comment:
This seems like an interesting tool, but I'm going to look into it a bit more before I feel comfortable adding it. At a high level I'm not sure if mocks (& asserting that they are called) is the right framework. We don't care how they are interacted with, but instead care a lot about the values that they return.

Reply:
Agree - I'm open to changing this pr up **a lot**. It's very much a PoC as opposed to a mergable - though we will need to test once we have a fix for flushing the batcher on reorgs.